### PR TITLE
[Core/Pathfinding] Some improvements... hopefully...

### LINKED ISSUE #862: Pit of Saron - Trash mobs problem
**Description**: Mobs after second boss, that come for event to 'block' free passage, drop under ground every single time.

**Current behaviour**: For some reason, they drop under texture, and keep agrro players from there.

**Expected behaviour**: They should stay at the top, so they can attack and kill them. Unable due LoS.

**Steps to reproduce the problem**: Not sure.

**Hash**: 7696bd3

**Operating system**:  Linux, Debian 9

**Modules**:  Premium, Congrats on level.

**Other Customizations**: No.

Pic: https://i.imgur.com/fD27U9Z.jpg

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/58227594-pit-of-saron-trash-mobs-problem?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #948: Kel'Thuzad Phase 1 undermap mobs
Description: Kel'Thuzad Phase 1 undermap mobs

Current behaviour: During Phase 1 in Kel'thuzad fight , mobs who appears in doodad_kelthuzad_window_portal01 and doodad_kelthuzad_window_portal02 are often in Undermap when  they move to targeted player , so players can't kill them.

Expected behaviour:  I think that's wrong position on waypoints in the mobs way or wrong position spawn under the ground.

Steps to reproduce the problem:

1.Go to Naxxramas , Kel'Thuzad last boss
2. Engage Kel'thuzad enter in the blue round 
3. Wait adds appear in the Portal 01 and 02 at the left of Kel'Thuzad when we are facing him.

Branch(es): master

AC hash/commit: 7329ad4

Operating system: Linux

Modules: /

Other Customizations: /

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/60088908-kel-thuzad-phase-1-undermap-mobs?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2124: Pit of Saron Normal/Heroic Dungeon - Ymirjar Deathbringer witches sometimes are stuck underground. [$5]
Pit of Saron Normal/Heroic Dungeon - Ymirjar Deathbringer witches sometimes are stuck underground.

##### SMALL DESCRIPTION:

Pit of Saron Normal/Heroic Dungeon - Ymirjar Deathbringer witches sometimes are stuck underground. If they aren't underground and when you pull them they will move and go underground. (check minimap below for exact spot)

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Ymirjar Deathbringer witches shouldn't be stuck underground.

##### CURRENT BEHAVIOUR:

80% of the runs. Two Ymirjar Deathbringer witches are always underground on this spot. If they aren't underground and when you pull them they will move and go underground. So players have to to shoot them through the ground. (check minimap)

minimap2
minimap

A couple more withes here, precise location minimap:
minimap

##### STEPS TO REPRODUCE THE PROBLEM:

1. Do a dungeon run of Pit of Saron Normal/Heroic
2. And find the Two Ymirjar Deathbringer witches (minimap check above)

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

3b09bd7 

##### OPERATING SYSTEM:

Ubuntu 16/18

##### OTHER CUSTOMIZATIONS:

<bountysource-plugin>

---
There is a **[$5 open bounty](https://www.bountysource.com/issues/77413210-pit-of-saron-normal-heroic-dungeon-ymirjar-deathbringer-witches-sometimes-are-stuck-underground?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2688: Grounded NPC flying over gap and crossing boundaries (mmap issue) [$75]
##### SMALL DESCRIPTION:

Mobs can fly over a gap, or climb up/down a wall while it's impossible normally.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

NPCs should follow the laws of nature, like the player characters do.

##### CURRENT BEHAVIOUR:

Mobs can fly over a gap, climb up/down a mountain or a wall etc... it's gamebreaking and visually horrible and screams "BUG" to the players' ears.

##### STEPS TO REPRODUCE THE PROBLEM:

1. `.cheat god`
2. `.go cr 90632` (inside BRD)
3. Attack this mob but do not kill it, at around 20% he will flee and join  the other side on the right and will pull all the mobs which is normally impossible and will wipe your group. This is kinda gamebreaking (and ugly to see).
4. You can also jump below and they will all follow you.

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

8760a9f5c9739726cda09a96c44c11d4a6813e72

##### OPERATING SYSTEM:

Deb 9

<bountysource-plugin>

---
There is a **[$75 open bounty](https://www.bountysource.com/issues/88852907-grounded-npc-flying-over-gap-and-crossing-boundaries-mmap-issue?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2912: Creatures and not only dropping underground after i use spells like Typhoon, Blink (Mage) or Thunderstorm (Sham)
Creatures and not only dropping underground after i use spells like Typhoon, Blink (Mage) or Thunderstorm (Sham)

##### SMALL DESCRIPTION:

Hello everyone, there is such a strange problem , creatures and not only dropping underground after i use spells like Typhoon, Blink (Mage) or Thunderstorm (Sham), also sometimes some aoe spells like Starfall pulling creature from another room and another floor. im looking for help about this sotuation. here is example in icecrown citadel using Typhoon
https://youtu.be/76yqrXYTXPs

##### STEPS TO REPRODUCE THE PROBLEM:

1. 
2. 
3. 

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

50287c05f0e3ac4d8f006543eb916d53efa15f5f

##### OPERATING SYSTEM:

Debian 9

##### OTHER CUSTOMIZATIONS:

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/91199127-creatures-and-not-only-dropping-underground-after-i-use-spells-like-typhoon-blink-mage-or-thunderstorm-sham?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by FALL1N1**
# CHANGES PROPOSED:
- Reduce "Max Walkable Angle" to 60*
- Use pathfinding for Point movement.
- Reduce fleeing movement (Not In LoS) check to 100ms

## ISSUES ADDRESSED:
- Closes #2688
- Closes #2912 
- Closes #862 (Potentially)
- Closes #948 (Potentially)
- Closes #2124(Potentially) 

## TESTS PERFORMED:
Below.

## HOW TO TEST THE CHANGES:
As of the issues, example:

```
    .cheat god
    .go cr 90632 (inside BRD)
    Attack this mob but do not kill it, at around 20% he will flee and join the other side on the right and will pull all the mobs which is normally impossible and will wipe your group. This is kinda gamebreaking (and ugly to see).
    You can also jump below and they will all follow you.
```

## KNOWN ISSUES AND TODO LIST:
None that I know of...

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FALL1N1** [comment]
If you are lazy to build it on your own, here are binaries and extracted mmaps/vmaps:

http://fs.kartselyanski.com/res.zip - dbc/maps/vmaps/mmaps
http://fs.kartselyanski.com/bin.zip - auth/worldserver together with all their stuff

**Comment by aradep** [comment]
Fleeing works better, but the group on the opposite side still follows through the floor.
https://www.youtube.com/watch?v=wDZ133abAuw

**Comment by FALL1N1** [comment]
Thank you, on it.

**Comment by Kitzunu** [comment]
What would this fix?
"Reduce "Max Walkable Angle" to 60*"

**Comment by FALL1N1** [comment]
@Kitzunu https://prnt.sc/vc4ga8 - red is current behaviour, wisps - with 60*

**Comment by FALL1N1** [comment]
Issue fixed, although do not merge before I clean everything up. http://prntscr.com/vfmops

**Comment by FrancescoBorzi** [comment]
Nice job @FALL1N1 ! Keep up the good work.

There is a compile error:

```
azerothcore-wotlk/src/server/game/Entities/Creature/Creature.cpp:166:90: fatal error: field 'm_cannotReachTimer' will be initialized after field 'm_AI_locked' [-Wreorder-ctor]
````

File: src/server/game/AI/CreatureAI.cpp [hunk 1]
```diff
@@ -249,6 +249,7 @@ bool CreatureAI::_EnterEvadeMode()
     me->SetLootRecipient(nullptr);
     me->ResetPlayerDamageReq();
     me->SetLastDamagedTime(0);
+	me->SetCannotReachTarget(false);
```
Comment (FrancescoBorzi) [inline src/server/game/AI/CreatureAI.cpp:252]: please fix indentation

File: src/server/game/Entities/Creature/Creature.cpp [hunk 2]
```diff
@@ -610,6 +610,15 @@ void Creature::Update(uint32 diff)
...
                 }
+
+                if (CanNotReachTarget() && !IsInEvadeMode() && !GetMap()->IsRaid())
+                {
+                    m_cannotReachTimer += diff;
+                    if (m_cannotReachTimer >= 5000) // 5s?
+                        if (IsAIEnabled)
+                            AI()->EnterEvadeMode();
+                } 
+
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Creature/Creature.cpp:621]: please move that `5000` to some constant, also the `if` could be rewritten as:

```cpp
if ( (m_cannotReachTimer >= SOME_CONST && IsAiEnabled)
{
   AI()->EnterEvadeMode();
}
```

File: src/server/game/Entities/Creature/Creature.h [hunk 3]
```diff
@@ -499,6 +499,7 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
     uint8 getLevelForTarget(WorldObject const* target) const override; // overwrite Unit::getLevelForTarget for boss level support
 
     bool IsInEvadeMode() const { return HasUnitState(UNIT_STATE_EVADE); }
+	bool IsEvadingAttacks() const { return IsInEvadeMode() || CanNotReachTarget(); }
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Creature/Creature.h:502]: indentation

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 4]
```diff
@@ -39,16 +39,18 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
...
-    bool forceDest = (owner->FindMap() && owner->FindMap()->IsDungeon() && !isPlayerPet) || // force in instances to prevent exploiting
+    bool forceDest = false; /* (owner->FindMap() && owner->FindMap()->IsDungeon() && !isPlayerPet) || // force in instances to prevent exploiting
                      (owner->GetTypeId() == TYPEID_UNIT && ((owner->IsPet() && owner->HasUnitState(UNIT_STATE_FOLLOW)) || // allow pets following their master to cheat while generating paths
                              ((Creature*)owner)->isWorldBoss() || ((Creature*)owner)->IsDungeonBoss())) || // force for all bosses, even not in instances
                      (owner->GetMapId() == 572 && (owner->GetPositionX() < 1275.0f || i_target->GetPositionX() < 1275.0f)) || // pussywizard: Ruins of Lordaeron - special case (acid)
                      sameTransport || // nothing to comment, can't find path on transports so allow it
-                     (i_target->GetTypeId() == TYPEID_PLAYER && i_target->ToPlayer()->IsGameMaster()); // for .npc follow
+                     (i_target->GetTypeId() == TYPEID_PLAYER && i_target->ToPlayer()->IsGameMaster()); // for .npc follow*/
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:47]: if we don't need this code anymore, it's better to remove it completely

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 5]
```diff
@@ -39,16 +39,18 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
...
     bool forcePoint = ((!isPlayerPet || owner->GetMapId() == 618) && (forceDest || !useMMaps)) || sameTransport;
 
-    if (owner->GetTypeId() == TYPEID_UNIT && !i_target->isInAccessiblePlaceFor(owner->ToCreature()) && !sameTransport && !forceDest && !forcePoint)
+    if (owner->GetTypeId() == TYPEID_UNIT && !i_target->isInAccessiblePlaceFor(owner->ToCreature()) && !sameTransport && !forceDest && !forcePoint) {
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:50]: the opening bracket should be on a new line for consistency

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 6]
```diff
@@ -186,6 +188,10 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
             float maxDist = MELEE_RANGE + owner->GetMeleeReach() + i_target->GetMeleeReach();
             if (!forceDest && (i_path->GetPathType() & PATHFIND_NOPATH || (!i_offset && !isPlayerPet && i_target->GetExactDistSq(i_path->GetActualEndPosition().x, i_path->GetActualEndPosition().y, i_path->GetActualEndPosition().z) > maxDist * maxDist)))
             {
+				// unsure, @todo
+				if (owner->GetTypeId() == TYPEID_UNIT)
+					owner->ToCreature()->SetCannotReachTarget(false);
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:193]: please add brackets

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 7]
```diff
@@ -202,9 +208,12 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
                 init.Launch();
                 return;
             }
-        }
-
-        // if failed to generate, just use normal MoveTo
+        } else {
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:211]: same as above (codestyle consistency)

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 8]
```diff
@@ -202,9 +208,12 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
...
+			// evade first
+			if (owner->GetTypeId() == TYPEID_UNIT)
+				owner->ToCreature()->SetCannotReachTarget(true);
+			// then use normal MoveTo
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:215]: please add brackets

File: src/tools/mmaps_generator/MapBuilder.cpp [hunk 9]
```diff
@@ -603,9 +603,7 @@ namespace MMAP
         config.borderSize = config.walkableRadius + 3;
         config.maxEdgeLen = VERTEX_PER_TILE + 1;        // anything bigger than tileSize
         config.walkableHeight = m_bigBaseUnit ? 3 : 6;
-        // a value >= 3|6 allows npcs to walk over some fences
-        // a value >= 4|8 allows npcs to walk over all fences
-        config.walkableClimb = m_bigBaseUnit ? 4 : 8;
+        config.walkableClimb = m_bigBaseUnit ? 3 : 6;
```
Comment (FrancescoBorzi) [inline src/tools/mmaps_generator/MapBuilder.cpp:606]: maybe keep some comments that explain this?

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp (hunk 4 above)
Comment (FALL1N1) [reply src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:47]: we do.. at least partially, it's pending rewrite and re-checking within the next couple of hours

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp (hunk 6 above)
Comment (FALL1N1) [reply src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:193]: not necessary for single statement/line

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp (hunk 8 above)
Comment (FALL1N1) [reply src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:215]: not necessary for single statement/line

File: src/tools/mmaps_generator/MapBuilder.cpp (hunk 9 above)
Comment (FALL1N1) [reply src/tools/mmaps_generator/MapBuilder.cpp:606]: I doubt we'd ever have to edit those again, but yeah, I have reverted it for reference.

File: src/server/game/Entities/Creature/Creature.cpp (hunk 2 above)
Comment (FALL1N1) [reply src/server/game/Entities/Creature/Creature.cpp:621]: thanks for the catch, fixed

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp (hunk 4 above)
Comment (FALL1N1) [reply src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:47]: didn't had to rewrite it, just commented out the bad stuff

**Comment by FALL1N1** [comment]
everything should be fine now, let's see if linux stops bitching about the position

**Comment by FrancescoBorzi** [comment]
@FALL1N1  still erroring:

```
src/server/game/Entities/Creature/Creature.cpp:169:153: fatal error: field '_lastDamagedTime' will be initialized after field 'm_cannotReachTarget' [-Wreorder-ctor]
    m_homePosition(), m_transportHomePosition(), m_creatureInfo(nullptr), m_creatureData(nullptr), m_waypointID(0), m_path_id(0), m_formation(nullptr), _lastDamagedTime(0), m_cannotReachTarget(false), m_cannotReachTimer(0)
```

**Comment by FALL1N1** [comment]
yeah, I saw that, sadly I cannot reproduce on windows, setting up my centos vm and will fix it later

File: src/server/game/Entities/Creature/Creature.cpp [hunk 10]
```diff
@@ -47,6 +47,8 @@
 #include "LuaEngine.h"
 #endif
 
+uint32 DEF_CANNOT_REACH = 5 * IN_MILLISECONDS; // 5s ok?
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Creature/Creature.cpp]:
```cpp
constexpr uint32 DEF_CANNOT_REACH = 5 * IN_MILLISECONDS
```

**Comment by FrancescoBorzi** [comment]
compile looks good now, waiting for testers feedback @azerothcore/testers

File: src/server/game/Entities/Creature/Creature.cpp [hunk 11]
```diff
@@ -610,6 +612,14 @@ void Creature::Update(uint32 diff)
 
                     m_regenTimer += CREATURE_REGEN_INTERVAL;
                 }
+
+                if (CanNotReachTarget() && !IsInEvadeMode() && !GetMap()->IsRaid())
+                {
+                    m_cannotReachTimer += diff;
+                    if (m_cannotReachTimer >= DEF_CANNOT_REACH && IsAIEnabled)
+                        AI()->EnterEvadeMode();
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Creature/Creature.cpp:620]: brackets please

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 12]
```diff
@@ -39,16 +39,21 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
...
     bool useMMaps = MMAP::MMapFactory::IsPathfindingEnabled(owner->FindMap()) && !sameTransport;
-    bool forceDest = (owner->FindMap() && owner->FindMap()->IsDungeon() && !isPlayerPet) || // force in instances to prevent exploiting
-                     (owner->GetTypeId() == TYPEID_UNIT && ((owner->IsPet() && owner->HasUnitState(UNIT_STATE_FOLLOW)) || // allow pets following their master to cheat while generating paths
-                             ((Creature*)owner)->isWorldBoss() || ((Creature*)owner)->IsDungeonBoss())) || // force for all bosses, even not in instances
+    bool forceDest =
+                     // (owner->FindMap() && owner->FindMap()->IsDungeon() && !isPlayerPet) || // force in instances to prevent exploiting
+                     (owner->GetTypeId() == TYPEID_UNIT && ((owner->IsPet() && owner->HasUnitState(UNIT_STATE_FOLLOW)))) || // allow pets following their master to cheat while generating paths
+                     // ((Creature*)owner)->isWorldBoss() || ((Creature*)owner)->IsDungeonBoss())) || // force for all bosses, even not in instances
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:45]: shall we remove the commented code?

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 13]
```diff
@@ -186,6 +191,10 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
             float maxDist = MELEE_RANGE + owner->GetMeleeReach() + i_target->GetMeleeReach();
             if (!forceDest && (i_path->GetPathType() & PATHFIND_NOPATH || (!i_offset && !isPlayerPet && i_target->GetExactDistSq(i_path->GetActualEndPosition().x, i_path->GetActualEndPosition().y, i_path->GetActualEndPosition().z) > maxDist * maxDist)))
             {
+                // unsure, @todo
+                if (owner->GetTypeId() == TYPEID_UNIT)
+                    owner->ToCreature()->SetCannotReachTarget(false);
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:196]: brackets please, remove the `TODO` if you are sure

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp [hunk 14]
```diff
@@ -203,8 +212,13 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool in
                 return;
             }
         }
-
-        // if failed to generate, just use normal MoveTo
+        else
+        {
+            // evade first
+            if (owner->GetTypeId() == TYPEID_UNIT)
+                owner->ToCreature()->SetCannotReachTarget(true);
```
Comment (FrancescoBorzi) [inline src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:219]: brackets please

**Review by FrancescoBorzi — COMMENTED** [review]
@FALL1N1 some minor codestyle remarks

**Comment by locus313** [comment]
This looks pretty good now. 
when standing on the corner the mobs can attack you, but don't fly over the gap anymore:
image
image
image
image
even when jumping down, they no longer fall through the floor:
image
image

File: src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp (hunk 12 above)
Comment (FALL1N1) [reply src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp:45]: let it stay for now, I have also added a comment below on what it's used for

"// closes "bool forceDest", that way it is more appropriate, so we can comment out crap whenever we need to"

**Comment by FALL1N1** [comment]
codestyle fixed, all good to go

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@aradep would you like to give this a final test ?

**Comment by FALL1N1** [comment]
I have tested it on both linux vm as well as windows, everything seems fine, but someone else should test it as well just in case.

**Comment by FALL1N1** [comment]
update: DO NOT MERGE yet, there's an issue and most of the dungeons/raids' bosses/mobs are in evade loop, currently working on it...

**Comment by FrancescoBorzi** [comment]
@FALL1N1 thanks for the update. I've just updated your branch with latest master, that includes a small change to the MMAPS system (https://github.com/azerothcore/azerothcore-wotlk/commit/2c20a869299e34bf994466000912f3c1e99b6ed1).

I suggest to update your branch locally and re-extract the maps before further development/testing.

**Comment by FALL1N1** [comment]
Actually, the evade issue was unrelated, can someone else test it as well?

Also, the merge with master does not require map/vmap/mmap re-extraction as it's unrelated, it can work with those as well.

**Comment by BarbzYHOOL** [comment]
I can't test sadly (while I opened the issue -__- what a shame) but i have few questions

If mobs attack from afar, do they start chasing you all over the dungeon though? (i think blizzlike?)
Or do they stand here waiting? (acceptable imo xD)
And what happens when you finally join them?

**Comment by FALL1N1** [comment]
@BarbzYHOOL they will evade, because they won't be able to find walkable path to their target.

Running to the player through the entire dungeon and pulling everything around will be blizzlike, but it's not supported with azeroth/trinity as far as I know, either way it's an issue for another day unrelated to the current one.

**Comment by aradep** [comment]
Tested, works great!
https://youtu.be/upwws6mVRvo

>http://fs.kartselyanski.com/res.zip - dbc/maps/vmaps/mmaps

👏  @FALL1N1!  🚀

**Comment by Voxstrasza** [comment]
i played around with this and it all seems good, nice work.

**Comment by FALL1N1** [comment]
Thanks for the tests guys, anyone else experiencing any issues?

**Comment by BarbzYHOOL** [comment]
"Running to the player through the entire dungeon and pulling everything around will be blizzlike, but it's not supported with azeroth/trinity as far as I know"

didn't know that! nobody ever told me so, requires some testing

@winfidonarleyan did you test the PR or on a live server?

**Comment by Winfidonarleyan** [comment]
No test, no now time

**Comment by Viste** [comment]
Are you sure this will not cause problems with creeps going into evade in some specific cases?

**Comment by FALL1N1** [comment]
@Viste as long as they have valid path to the player they won't have issues, it's blizzlike behavior so anything else than that is MMap issue, either incorrect extraction data or something else.

Although, if there are - we can add a custom hack (exception) like the current one that gets reverted by Sunwell's devs.

**Comment by BarbzYHOOL** [comment]
Anyone still using this regularly @locus313 @aradep @Voxstrasza ? no bad feedback if so?

**Comment by locus313** [comment]
been using it regularly for the past couple weeks, its been working great.

**Comment by BarbzYHOOL** [comment]
ok imo we can merge it, dunno what others think about that

**Comment by Helias** [comment]
Have been tested all the cases of the related issues?
```
Closes #2688
Closes #2912
Closes #862 (Potentially)
Closes #948 (Potentially)
Closes #2124(Potentially)
```

**Comment by LukasVolgger** [comment]
Tested this PR - everything works perfect!

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2688
image

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/862 
image

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2124
image

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2912
Played with many creatures and used blink as mage - no drops under the floor

- Issue https://github.com/azerothcore/azerothcore-wotlk/issues/948 already closed by me

**Comment by LukasVolgger** [comment]
@FALL1N1 Awesome work, thx! Can this be merged?

**Comment by Helias** [comment]
Good job guys :rocket:

**Review by Si1ker — APPROVED** [review]

**Comment by boom8866** [comment]
Have met, creatures in the underground city, can randomly out of combat?

**Comment by boom8866** [comment]
@FALL1N1 

Please reopen this question, there is a problem with the update, which leads to an error in the judgment of mmasp

1. .go  641.640 191.140 511.736 658  Through this Stone, creatures can quickly get out of combat

2.The premise is that you need a melee creature, guide him here, and then jump to the top of Stone, and you will find that the creature is out of combat

I have used the latest version of the update[Map file re-extraction]

**Comment by FrancescoBorzi** [comment]
@boom8866 can you please open a new issue report?

**Comment by boom8866** [comment]
ok

**Pull request merged**